fontbe/src/mvar.rs

```diff
@@ -0,0 +1,329 @@
+            if let Some(mvar_tag) = metric.mvar_tag() {
+                mvar_builder.add_sources(mvar_tag, values)?;
+            }
+        }
+        let mvar = mvar_builder.build();
+
+        context.mvar.set_unconditionally(mvar.into());
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
+    use fontdrasil::{
+        coords::{CoordConverter, UserCoord},
+        types::Axis,
+    };
+    use write_fonts::{
+        dump_table,
+        read::{
+            tables::{mvar as read_mvar, variations::ItemVariationData},
+            FontData, FontRead,
+        },
+        types::F2Dot14,
+    };
+
+    use super::*;
+
+    fn axis(tag: &str, min: f32, default: f32, max: f32) -> Axis {
+        let min = UserCoord::new(min);
+        let default = UserCoord::new(default);
+        let max = UserCoord::new(max);
+        Axis {
+            name: tag.to_string(),
+            tag: Tag::from_str(tag).unwrap(),
+            min,
+            default,
+            max,
+            hidden: false,
+            converter: CoordConverter::unmapped(min, default, max),
+        }
+    }
+
+    fn new_mvar_builder(locations: Vec<&NormalizedLocation>, axes: Vec<Axis>) -> MvarBuilder {
+        let locations = locations.into_iter().cloned().collect();
+        let model = VariationModel::new(locations, axes).unwrap();
+        MvarBuilder::new(model)
+    }
+
+    fn add_sources(
+        builder: &mut MvarBuilder,
+        mvar_tag: &str,
+        sources: &[(&NormalizedLocation, f32)],
+    ) {
+        let sources = sources
+            .iter()
+            .map(|(loc, value)| ((*loc).clone(), (*value).into()))
+            .collect::<HashMap<_, _>>();
+        builder
+            .add_sources(Tag::from_str(mvar_tag).unwrap(), &sources)
+            .unwrap();
+    }
+
+    fn delta_sets(var_data: &ItemVariationData) -> Vec<Vec<i32>> {
+        (0..var_data.item_count())
+            .map(|i| var_data.delta_set(i).collect::<Vec<_>>())
+            .collect()
+    }
+
+    #[test]
+    fn smoke_test() {
+        let regular = NormalizedLocation::for_pos(&[("wght", 0.0)]);
+        let bold = NormalizedLocation::for_pos(&[("wght", 1.0)]);
+        let mut builder = new_mvar_builder(
+            vec![&regular, &bold],
+            vec![axis("wght", 400.0, 400.0, 700.0)],
+        );
+
+        add_sources(&mut builder, "xhgt", &[(&regular, 500.0), (&bold, 550.0)]);
+
+        let Some(mvar) = builder.build() else {
+            panic!("no MVAR?!");
+        };
+
+        let bytes = dump_table(&mvar).unwrap();
+        let mvar = read_mvar::Mvar::read(FontData::new(&bytes)).unwrap();
+
+        assert_eq!(mvar.version(), MajorMinor::VERSION_1_0);
+        assert_eq!(mvar.value_records().len(), 1);
+
+        let rec = &mvar.value_records()[0];
+        assert_eq!(rec.value_tag(), Tag::new(b"xhgt"));
+        assert_eq!(rec.delta_set_outer_index(), 0);
+        assert_eq!(rec.delta_set_inner_index(), 0);
```

Comment:
Optional, I continue to greatly prefer the test output of this sort of thing done as one assert, e.g.

```rust
assert_eq!(vec![(Tag::new(b"xhgt"), 0, 0),], mvar.value_records().iter().map(...)]
```

I've always found it much easier to troubleshoot and harder to forget to assert the length